lib/nerves_network/udhcpc.ex

```diff
@@ -133,11 +133,8 @@ defmodule Nerves.Network.Udhcpc do
   end
 
   defp hostname() do
-    # Turn :sname@host into a hostname
-    # Returns "nohost" if Erlang distribution not enabled
-    node()
-      |> to_string
-      |> String.split("@")
-      |> Enum.at(1)
+    {:ok, hostname} = :inet.gethostname()
```

Comment:
This won't return `"nohost"` anymore, so you can probably simplify the code up on L62, L68-69.

I'm not sure under what circumstances it would fail and return something other than the hostname, but it might be better to somehow still function but not pass the hostname to the DHCP server in that case.